here-naksha-lib-model/src/commonMain/kotlin/naksha/model/NakshaCollectionProxy.kt

```diff
@@ -0,0 +1,145 @@
+     * Note that if disableHistory is false, the deleted features will still be around in the history. This mainly effects lib-view.
+     */
+    var autoPurge: Boolean by AUTO_PURGE
+
+    /**
+     * The indices list contains the list of indices to add to the collection.
+     * If set to null, default indices are created.
+     * The available indices are exposed through the virtual table naksha~indices.
+     */
+    var indices: naksha.model.IndicesListProxy by INDICES
+
+    /**
+     * The maxAge decides about the maximum age of features in the history in days.
+     * Note that there is no guarantee that features are deleted exactly after having reached their max-age.
+     * However, they are eligible to be deleted at as soon as possible.
+     */
+    var maxAge: Int64 by MAX_AGE
+
+    /**
+     * The quadPartitionSize decides (for the optimal partitioning algorithm) how many features should be placed into each "optimal" tile.
+     */
+    var quadPartitionSize: Int by QUAD_PARTITION_SIZE
+
+    var estimatedFeatureCount: Int64? by ESTIMATED_FEATURE_COUNT
+
+    var estimatedDeletedFeatures: Int64? by ESTIMATED_DELETED_FEATURES
+
+    companion object {
+        private val PARTITIONS = NotNullProperty<Any, com.here.naksha.lib.base.GeoFeature, Int>(Int::class, 1)
+        private val GEO_INDEX = NullableProperty<Any, com.here.naksha.lib.base.GeoFeature, String>(String::class, defaultValue = "gist")
+        private val STORAGE_CLASS = NullableProperty<Any, com.here.naksha.lib.base.GeoFeature, String>(String::class)
+        private val PROTECTION_CLASS = NullableProperty<Any, com.here.naksha.lib.base.GeoFeature, String>(String::class)
+        private val DEFAULT_TYPE = NotNullProperty<Any, com.here.naksha.lib.base.GeoFeature, String>(String::class, defaultValue = "Feature")
+        private val DEFAULT_FLAGS =
+            NotNullProperty<Any, com.here.naksha.lib.base.GeoFeature, Int>(Int::class, defaultValue = naksha.model.Flags.DEFAULT_FLAGS)
+        private val DISABLE_HISTORY = NotNullProperty<Any, com.here.naksha.lib.base.GeoFeature, Boolean>(Boolean::class, false)
+        private val AUTO_PURGE = NotNullProperty<Any, com.here.naksha.lib.base.GeoFeature, Boolean>(Boolean::class, false)
+        private val INDICES = NotNullProperty<Any, com.here.naksha.lib.base.GeoFeature, naksha.model.IndicesListProxy>(
+            naksha.model.IndicesListProxy::class)
+        private val MAX_AGE = NotNullProperty<Any, com.here.naksha.lib.base.GeoFeature, Int64>(Int64::class, defaultValue = Int64(-1))
+        private val QUAD_PARTITION_SIZE = NotNullProperty<Any, com.here.naksha.lib.base.GeoFeature, Int>(Int::class, defaultValue = 10_485_760)
+        private val ESTIMATED_FEATURE_COUNT = NullableProperty<Any, com.here.naksha.lib.base.GeoFeature, Int64>(Int64::class)
+        private val ESTIMATED_DELETED_FEATURES = NullableProperty<Any, com.here.naksha.lib.base.GeoFeature, Int64>(Int64::class)
+
+
```

Comment:
We should make this a secondary constructor and simply annotate it with for example `@JsName("of")`, this is natural in Kotlin and Java (`[new] NakshaCollectionProxy(id, part, ...)`) and in JavaScript the same can be done via `NakshaCollectionProxy.of(id, part, ...)`